### PR TITLE
Add PTB and Canary versions

### PR DESCRIPTION
**Pull request by Gabriel29306**
# Description
This PR permit to satisfy the #8 issue.
This PR reformat the code to be reusable between different versions of discord.

# Modifications
- Rename the `packages.json` file in `stable_packages.json`.
- Add `ptb_packages.json` and `canary_packages.json` for the differents versions.
- Transform the main code to a function to be reused.
- Use the function for all versions.

> [!Note]
> If needed, I can modify the code as you need.

### CODE REVIEW

File: get_new_package.py [hunk 1]
```diff
@@ -4,39 +4,75 @@
...
+                continue
+            print("getting", package_name)
+            with requests.get(package_url, stream=True) as r:
+                local_path = debian.joinpath("pool", package_name)
+                with local_path.open('wb') as f:
+                    shutil.copyfileobj(r.raw, f)
+        return 0
+    else:
+        print(f"Already have {filename}", url)
+        return 1
```
Comment (palfrey) [inline get_new_package.py:41]: The return 0/1 was an exit code. For this version, it should return a bool from this function and if any of the calls to this return true, return 0 else 1

File: get_new_package.py [hunk 2]
```diff
@@ -4,39 +4,75 @@
...
+
+
+# For Stable Version
+####################
+stable_initial_req = requests.get(
+    "https://discord.com/api/download?platform=linux&format=deb", allow_redirects=False)
+stable_initial_req.raise_for_status()
+
+stable_url = stable_initial_req.headers["Location"]
+stable_filename = stable_url.split("/")[-1]
```
Comment (palfrey) [inline get_new_package.py:51]: Other than the varying URL, I'm seeing the same code here for stable/ptb/canary which could just be folded into one big "download stuff" function

**Review by palfrey — CHANGES_REQUESTED** [review]
Looks pretty good. There's a couple of comments below, and also the README will need updating to note the 3 different packages, but happy to merge if those can get resolved.

**Comment by Gabriel29306** [comment]
Is it good for you or you need a other change ?

**Comment by palfrey** [comment]
> Is it good for you or you need a other change ?

Can you please fix the conflict with `packages.json` - I think the stable list just updated since you branched for this, and then I'll merge.

**Comment by Gabriel29306** [comment]
Now it's normally good.

**Comment by Gabriel29306** [comment]
There is a problem with my PR. But i don't know where the problem is. I'm searching th problem source.

**Comment by Gabriel29306** [comment]
I think that because the `debian/Packages` is writted with some problems, there is a alternance between stable and Canary + PTB

**Pull request merged**